src/main/kotlin/org/jetbrains/research/ml/ast/transformations/augmentedAssignment/AugmentedAssignmentTransformation.kt

```diff
@@ -0,0 +1,26 @@
+package org.jetbrains.research.ml.ast.transformations.augmentedAssignment
+
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import com.jetbrains.python.psi.PyAugAssignmentStatement
+import org.jetbrains.research.ml.ast.transformations.Transformation
+
+class AugmentedAssignmentTransformation : Transformation {
+    override val metadataKey: String = "AugmentedAssignment"
+
+    override fun apply(psiTree: PsiElement, toStoreMetadata: Boolean) {
+        val augStatements = PsiTreeUtil.collectElementsOfType(psiTree, PyAugAssignmentStatement::class.java)
+
+        val visitor = AugmentedAssignmentVisitor()
+        WriteCommandAction.runWriteCommandAction(psiTree.project) {
+            for (statement in augStatements) {
+                statement.accept(visitor)
+            }
+        }
```

Comment:
You use this code in all transformations. Maybe should we create the following function?
```
fun PyElementVisitor.acceptCollection(elements: Collection<PsiElement>, project: Project) {
        WriteCommandAction.runWriteCommandAction(project) {
            for (statement in elements) {
                statement.accept(this)
            }
        }
    }

    override fun apply(psiTree: PsiElement, toStoreMetadata: Boolean) {
        val augStatements = PsiTreeUtil.collectElementsOfType(psiTree, PyAugAssignmentStatement::class.java)
        val visitor = AugmentedAssignmentVisitor()
        visitor.acceptCollection(augStatements, psiTree.project)
    }
```

Reply:
Mb. But naming should me different. Because it not only "accepts" all elements from collection but doing it in write action

Reply:
Also as I noticed in IntelliJ SDK only the instance of PsiElement "accepts" visitor while PyElementVisitor "visits" the elements of AST.